File: py/examples/openai_example.py
#!/usr/bin/env python

from braintrust import init_logger, traced, wrap_openai
from openai import OpenAI

logger = init_logger(project="example-openai-project")
client = wrap_openai(OpenAI())


# @traced automatically logs the input (args) and output (return value)
# of this function to a span. To ensure the span is named `answer_question`,
# you should name the function `answer_question`.
@traced
def answer_question(body: str) -> str:
    prompt = [
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": body},
    ]

    result = client.chat.completions.create(
        model="gpt-4o",
        messages=prompt,
        temperature=0.5,
    )
    return result.choices[0].message.content


def main():
    input_text = "What's the capital of Australia?"
    result = answer_question(input_text)
    print(result)


if __name__ == "__main__":
    main()

File: js/examples/openai_example.ts
#!/usr/bin/env tsx

import OpenAI from "openai";
import { wrapOpenAI, initLogger } from "braintrust";

// Create and wrap OpenAI client with Braintrust
const openai = wrapOpenAI(new OpenAI());

initLogger({ projectName: "typescript-examples" });

async function main() {
  const result = await openai.chat.completions.create({
    messages: [
      {
        role: "user",
        content: "Hello! Can you tell me a joke?",
      },
    ],
    model: "gpt-3.5-turbo",
    max_tokens: 100,
  });
  console.dir(result);
}

main().catch(console.error);
